src/note.rs

```diff
@@ -113,10 +113,14 @@ impl Note {
         nonce: JubJubScalar,
         psk: &PublicSpendKey,
         value: u64,
-        blinding_factor: JubJubScalar,
+        mut blinding_factor: JubJubScalar,
```

Comment:
This shouldn't be mutable.
The `deterministic` function should accept the argument as is. We might have a `debug_assert!` to ensure the `blinding_factor` is actually zero, but that's the only thing we want to perform inside the `deterministic` method.

Reply:
Good point, fixed